client/components/main/index.jsx

```diff
@@ -18,6 +19,7 @@ export default function Main( {
 
 	return (
 		<main className={ classes } id={ id || null } role="main">
+			<ScrollToTopOnMount />
```

Comment:
I'm not convinced we want to apply this change everywhere where `<Main />` is used. 

For example, how can you be sure that `<Main />` won't be remounted while staying on the same screen and this would cause an unintended scroll to the top of the page?

Reply:
I agree that this PR was too ambitious and hard to test. The scenario where Main is re-mounted, and we will not want to scroll to the top, is very unlikely, but I agree that we need to ensure it won't cause any problems.